src/components/AllArticlesButtons.jsx

```diff
@@ -12,25 +12,29 @@ const AllArticlesButtons = ({ setSelectedTopic }) => {
     flexWrap: "wrap",
   };
 
-  function handleClick(event) {
-    setSelectedTopic(event.target.value);
-  }
-
   return (
     <>
       <ButtonGroup style={btnGrpStyle}>
-        <Button onClick={handleClick} variant="success" value="">
-          <Link to="/articles/all">All</Link>
-        </Button>
-        <Button onClick={handleClick} variant="warning" value="cooking">
-          <Link to="/articles/cooking">Cooking</Link>
-        </Button>
-        <Button onClick={handleClick} variant="info" value="coding">
-          <Link to="/articles/coding">Coding</Link>
-        </Button>
-        <Button onClick={handleClick} variant="danger" value="football">
-          <Link to="/articles/football"> Football</Link>
-        </Button>
+        <LinkContainer to="/articles/all">
+          <Button variant="success" value="">
+            All
+          </Button>
+        </LinkContainer>
+        <LinkContainer to="/articles/cooking">
+          <Button variant="warning" value="cooking">
+            Cooking
+          </Button>
+        </LinkContainer>
+        <LinkContainer to="/articles/coding">
+          <Button variant="info" value="coding">
+            Coding
+          </Button>
+        </LinkContainer>
+        <LinkContainer to="/articles/football">
+          <Button variant="danger" value="football">
+            Football
+          </Button>
+        </LinkContainer>
```

Comment:
You could populate this list of category links using a API request that fetches the categories. Imagine if someone added a new category to your website, you'd have to go in and manually change this menu - but if it used a list of categories fetched from the DB you wouldn't have to.
This isn't a manadatory change just something to consider!